src/models/history/tree-manager.ts

```diff
@@ -614,15 +615,17 @@ const historyEntryConverter = {
 };
 
 interface IPrepareFirestoreHistoryInfoArgs {
-  userContext?: IUserContext;
+  userContextProvider?: UserContextProvider;
   mainDocument?: IMainDocument;
   firestore?: Firestore;
 }
 
 async function prepareFirestoreHistoryInfo(
-    {userContext, mainDocument, firestore}: IPrepareFirestoreHistoryInfoArgs): Promise<IFirestoreHistoryInfo> {
+    {userContextProvider, mainDocument, firestore}: IPrepareFirestoreHistoryInfoArgs): Promise<IFirestoreHistoryInfo> {
+  // TODO: Wait for userContext to be valid.
```

Comment:
We never came back to this. It seems like it isn't a problem, but more a comment would be helpful:
```suggestion
  // TODO: Wait for userContext to be valid.
  //   the userContext initially starts out with a user id of 0, and it doesn't have a the portal and
  //   other properties defined. After the user is authenticated the userContext will have valid fields. 
  //   The invalid userContext will cause an error below.
  //   So we should use something like a MobX `await when(() => userContext?.uid)`.
  //   So far we haven't seen a case where prepareFirestoreHistoryInfo is called before the userContext 
  //   is ready. 
```